### DISCUSSION TITLE
TShock 4.5.11

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2520] [Show and tell]
## Notable changes
* Add the new allowed buff TentacleSpike to NPC buff cheat detection bouncer. (@sgkoishi)
* Changed hook `GetDataHandlers.OnNewProjectile` so that it passes the projectile's AI (by updating `NewProjectileEventArgs` and parsing this during the TShock hook) to support processing projectile AI in bouncer. (@AgaSpace)
* Fixed an issue where certain projectiles could be sent to the server with uncapped size parameters, which resulted in large disruptive client artifacts that could be used to grief players. (@AgaSpace, @Arthri)
* Added the currently running value of `Main.GameMode` to `/worldmode` as "Mode". (@hakusaro)

## Upgrading
To upgrade TShock, replace all of the files in the location of your server installation with all of the files from the release zip we provide. Overwrite all files. No data will be lost.

<hr /><em>This discussion was created from the release <a href='https://github.com/Pryaxis/TShock/releases/tag/v4.5.11'>TShock 4.5.11</a>.</em>

**Comment by Kojirremer** [comment]
``Added the currently running value of Main.GameMode to /worldmode as "Mode". (@hakusaro)``
Does that mean you can get the world difficulty using the command now? As in no more "which difficulty" being asked nonstop by players? :dagger:

**Reply by hakusaro** [reply]
Yes, but it's not very human friendly. It could probably be useful to add this to the MOTD, but I don't think this solves monster difficulty.
